src/momento/config/middleware/aio/middleware_metadata.py

```diff
@@ -0,0 +1,14 @@
+from typing import Optional
+
+from grpc.aio import Metadata
+
+
+class MiddlewareMetadata:
```

Comment:
Consider making this a dataclass. You'd have to lift `grpc_metadata` to be a class variable.